GetStream/stream-chat-android
# Improve QueryChannelsState::channels initial value

closes #3511 

### 🎯 Goal

Improve channels state handling.

### 🛠 Implementation details
`QueryChannelsList::channels` is being used together with `QueryChannelsList::loading` to map current data to `ChannelsStateData`. The problem is that both state flows are not synchronized and the `channels` value might not be changed yet when `loading` is already set to `false`.
`QueryChannelsList::channels` was initialized with an `emptyList` and because of that, we couldn't distinguish between "loading" and empty state and that produces "empty state view" flashing.

### 🧪 Testing

The issue is not 100% reproducible so run the test multiple times:
1. Open the sample app and sign in 
2. Observe the channel list and make sure `Empty State View` is not visible between loading and result list
3. Sign out or kill the app
4. Go to 1.

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
